docs/modules/migrate/pages/upgrading-from-jet.adoc

```diff
@@ -0,0 +1,417 @@
+
+{description}
+
+This guide is not an overview of the new features in Hazelcast {page-component-version}. This release includes other enhancements that don't need any special attention during an upgrade. For a complete list of changes, see the xref:ROOT:release-notes.adoc[release notes].
+
+== Before you Upgrade
+
+If you're using Jet Enterprise version 4.x, you can run your existing jobs on an upgraded cluster. Just cancel any running jobs on your Jet cluster before upgrading. Then, resubmit the job to the upgraded cluster.
+
+WARNING: You will lose the state of any running jobs that have not yet completed.
+
+For help canceling jobs, see xref:pipelines:job-management.adoc[].
+
+See xref:pipelines:submitting-jobs.adoc[].
+
+== Compatibility
+
+Hazelcast {page-component-version} is compatible only with the Job and Pipeline APIs of Jet 4.x. As a result, you can resubmit jobs that were running on Jet 4.x clusters to a {page-component-version} cluster. But, you *cannot* do the following:
+
+- Upgrade to {page-component-version} using rolling upgrades.
+- Connect to a {page-component-version} cluster using version 4.x of clients.
+- Resubmit jobs from snapshots that were made on Jet 4.x clusters.
+- Use versions of Management Center before {page-component-version}.
+
+== Jet Engine Security
+
+In embedded mode (Hazelcast in the same JVM as your application), you cannot use Jet APIs or SQL by default. These features are disabled for security reasons. For more information about security in the Jet engine and how to enable/disable features, see xref:pipelines:job-security.adoc[].
+
+== Configuration
+
+Hazelcast {page-component-version} includes important changes to Jet engine configuration.
+
+=== IMDG Configuration
+
+The following methods for configuring the underlying IMDG instance of Jet have been removed from the `JetConfig` object:
+
+- `getHazelcastConfig()`
+- `configureHazelcast()`
+- `setHazelcastConfig()`
+
+If you used the link:https://jet-start.sh/javadoc/4.5/com/hazelcast/jet/config/JetConfig.html[`JetConfig` object] to configure IMDG, you should replace instances of `JetConfig` with link:https://docs.hazelcast.org/docs/{page-component-version}/javadoc/com/hazelcast/config/Config.html[`Config`].
+
+[tabs] 
+==== 
+Jet 4.x:: 
++ 
+-- 
+```java
+JetConfig config = new JetConfig().configureHazelcast(c -> {
+  c.getNetworkConfig().setPort(8000);
+  c.setClusterName("jet-dev");
+  });
+
+JetInstance jet = Jet.newJetInstance(config);
+```
+--
+Hazelcast:: 
++ 
+-- 
+```java
+Config config = new Config() ;
+config.setPort(8000);
+config.setClusterName("jet-dev");
+
+HazelcastInstance hz = Hazelcast.newHazelcastInstance(config);
+```
+--
+====
+
+=== InstanceConfig
+
+The link:https://jet-start.sh/javadoc/4.5/com/hazelcast/jet/config/InstanceConfig.html[`InstanceConfig` object] has been removed and all its methods have been added to the `JetConfig` object.
```

Comment:
it's been deprecated and will be removed in a future release (it's not removed yet).